cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -992,115 +1108,123 @@ func TestStaticAutoscalerRunOnceWithFilteringOnUpcomingNodesEnabledNoScaleUp(t *
 }
 
 func TestStaticAutoscalerRunOnceWithSchedulerProcessingIgnored(t *testing.T) {
-	ignoredScheduler := "ignored-scheduler"
-	nonIgnoredScheduler := "non-ignored-scheduler"
-
-	readyNodeLister := kubernetes.NewTestNodeLister(nil)
-	allNodeLister := kubernetes.NewTestNodeLister(nil)
-	allPodListerMock := &podListerMock{}
-	podDisruptionBudgetListerMock := &podDisruptionBudgetListerMock{}
-	daemonSetListerMock := &daemonSetListerMock{}
-	onScaleUpMock := &onScaleUpMock{}
-	onScaleDownMock := &onScaleDownMock{}
-	deleteFinished := make(chan bool, 1)
-
-	now := time.Now()
-	later := now.Add(1 * time.Minute)
-
-	n1 := BuildTestNode("n1", 1000, 1000)
-	SetNodeReadyState(n1, true, time.Now())
-
-	p1 := BuildTestPod("p1", 600, 100)
-	p1.Spec.NodeName = "n1"
-	p2 := BuildTestPod("p2", 600, 100, MarkUnschedulable())
-	p3 := BuildTestPod("p3", 600, 100, AddSchedulerName(apiv1.DefaultSchedulerName)) // Not yet processed by scheduler, default scheduler is ignored
-	p4 := BuildTestPod("p4", 600, 100, AddSchedulerName(ignoredScheduler))           // non-default scheduler & ignored, expects a scale-up
-	p5 := BuildTestPod("p5", 600, 100, AddSchedulerName(nonIgnoredScheduler))        // non-default scheduler & not ignored, shouldn't cause a scale-up
-
-	provider := testprovider.NewTestCloudProvider(
-		func(id string, delta int) error {
-			return onScaleUpMock.ScaleUp(id, delta)
-		}, func(id string, name string) error {
-			ret := onScaleDownMock.ScaleDown(id, name)
-			deleteFinished <- true
-			return ret
-		})
-	provider.AddNodeGroup("ng1", 1, 10, 1)
-	provider.AddNode("ng1", n1)
-
-	ng1 := reflect.ValueOf(provider.GetNodeGroup("ng1")).Interface().(*testprovider.TestNodeGroup)
-	assert.NotNil(t, ng1)
-	assert.NotNil(t, provider)
-
-	// Create context with mocked lister registry.
+	bypassedScheduler := "bypassed-scheduler"
+	nonBypassedScheduler := "non-bypassed-scheduler"
 	options := config.AutoscalingOptions{
 		NodeGroupDefaults: config.NodeGroupAutoscalingOptions{
 			ScaleDownUnneededTime:         time.Minute,
 			ScaleDownUnreadyTime:          time.Minute,
 			ScaleDownUtilizationThreshold: 0.5,
 			MaxNodeProvisionTime:          10 * time.Second,
 		},
-		EstimatorName:             estimator.BinpackingEstimatorName,
-		ScaleDownEnabled:          true,
-		MaxNodesTotal:             10,
-		MaxCoresTotal:             10,
-		MaxMemoryTotal:            100000,
-		IgnoreSchedulerProcessing: true,
-		IgnoredSchedulers: map[string]bool{
+		EstimatorName:    estimator.BinpackingEstimatorName,
+		ScaleDownEnabled: true,
+		MaxNodesTotal:    10,
+		MaxCoresTotal:    10,
+		MaxMemoryTotal:   100000,
+		BypassedSchedulers: map[string]bool{
```

Comment:
Can you use `GetBypassedSchedulersMap` here? Right now this list contains default scheduler, but not empty scheduler name, which is a config we will never see in practice.

Reply:
Done